include/oneapi/dpl/pstl/execution_defs.h

```diff
@@ -123,27 +123,30 @@ inline constexpr unsequenced_policy unseq{};
 
 // 2.3, Execution policy type trait
 template <class T>
-struct is_execution_policy : ::std::false_type
+struct __is_execution_policy_impl : ::std::false_type
 {
 };
 
 template <>
-struct is_execution_policy<oneapi::dpl::execution::sequenced_policy> : ::std::true_type
+struct __is_execution_policy_impl<oneapi::dpl::execution::sequenced_policy> : ::std::true_type
 {
 };
 template <>
-struct is_execution_policy<oneapi::dpl::execution::parallel_policy> : ::std::true_type
+struct __is_execution_policy_impl<oneapi::dpl::execution::parallel_policy> : ::std::true_type
 {
 };
 template <>
-struct is_execution_policy<oneapi::dpl::execution::parallel_unsequenced_policy> : ::std::true_type
+struct __is_execution_policy_impl<oneapi::dpl::execution::parallel_unsequenced_policy> : ::std::true_type
 {
 };
 template <>
-struct is_execution_policy<oneapi::dpl::execution::unsequenced_policy> : ::std::true_type
+struct __is_execution_policy_impl<oneapi::dpl::execution::unsequenced_policy> : ::std::true_type
 {
 };
 
+template <typename T>
+using is_execution_policy = __is_execution_policy_impl<::std::decay_t<T>>;
+
```

Comment:
Since `is_execution_policy` is a public traits aligned with the C++ Standard definition, are we sure we need to consider cv- or ref- qualified execution policy classes as execution policies? Standard does not requires this trait to be true for any cv- and ref- qualifiers: https://timsong-cpp.github.io/cppwp/n4868/execpol.type. 
Microsoft implementation also defines it in the similar way as it was before this PR.